lib/ThreeScaleClient.php

```diff
@@ -279,11 +339,55 @@ public function authrep($appId, $appKey = null, $usage = null, $userId = null, $
     
   }
 
-  public function authrep_with_user_key($userKey, $usage = null, $userId = null, $object = null, $no_body = null, $serviceId = null) {  
+  /**
+   * Authorize and report in a single shot for user key auth mode
+   *
+   * @param $userKey  User key.
+   * @param $serviceId service id.
+   *
+   * @return ThreeScaleResponse object containing additional authorization information.
+   * If both provider key and application id are valid, the returned object is actually
+   * @see ThreeScaleAuthorizeResponse (which is derived from ThreeScaleResponse) and
+   * contains additional information about the usage status.
+   *
+   * @see ThreeScaleResponse
+   * @see ThreeScaleAuthorizeResponse
+   *
+   * @throws ThreeScaleServerError in case of unexpected internal server error
+   *
+   * Example:
+   *
+   * <code>
+   *   <?php
+   *   $response = $client->authorize('user_key');
+   *   // or $response = $client->authorize('user_key','service_id');
+   *
+   *   if ($response->isSuccess()) {
+   *     // ok.
+   *   } else {
+   *     // something is wrong.
+   *   }
+   *   ?>
+   * </code>
+   */
+
+  public function authrep_with_user_key($userKey, $serviceId = null, $usage = null, $userId = null, $object = null, $no_body = null) {  
```

Comment:
I am not sure how earlier changes were working. It always failed for me.

If I have $serviceId as last parameter```public function authrep_with_user_key($userKey, $usage = null, $userId = null, $object = null, $no_body = null, $serviceId = null) {```

It returns me error from ```throw new ThreeScaleServerError($httpResponse);```

The only way to fix was passing $serviceId as second parameter.

I have tested both provider key and service token with ```public function authrep_with_user_key($userKey, $serviceId = null, $usage = null, $userId = null, $object = null, $no_body = null) {  ``` and below are the working results

### Session token
$client =  new ThreeScaleClient("your_session_token", true);
$response = $client->authrep_with_user_key($_GET["user_key"], "service id");

### Provider Key 
//old users will be using
$client =  new ThreeScaleClient("your_provider_key");
$response = $client->authrep_with_user_key($_GET["user_key"], "service id");

If ```service_id``` is not passed then it will authenticate the key with default service